eessi/testsuite/tests/apps/MetalWalls.py

```diff
@@ -0,0 +1,103 @@
+see https://reframe-hpc.readthedocs.io/en/stable/manpage.html#test-filtering
+
+Hooks acting on all possible test combinations (before filtering) are called after the 'init' stage.
+Hooks acting on concrete test cases (after filtering) are called after the 'setup' stage.
+
+See also https://reframe-hpc.readthedocs.io/en/stable/pipeline.html
+"""
+import reframe as rfm
+from hpctestlib.sciapps.metalwalls.benchmarks import MetalWallsCheck
+from reframe.core.builtins import run_after
+from reframe.core.parameters import TestParam as parameter
+
+from eessi.testsuite import hooks
+from eessi.testsuite.constants import (COMPUTE_UNIT, CPU, DEVICE_TYPES, GPU,
+                                       SCALES, TAGS)
+from eessi.testsuite.utils import find_modules, log
+
+
+@rfm.simple_test
+class EESSI_MetalWalls_MW(MetalWallsCheck):
+    """MetalWalls benchmark tests.
+
+    `MetalWalls <https://gitlab.com/ampere2/metalwalls>`__ """
+
+    scale = parameter(SCALES.keys())
+
+    valid_systems = ['*']
+    valid_prog_environs = ['default']
+    time_limit = '30m'
+
+    module_name = parameter(find_modules('MetalWalls'))
+    # For now, MetalWalls is being build for CPU targets only
+    # compute_device = parameter([DEVICE_TYPES[CPU], DEVICE_TYPES[GPU]])
+    compute_device = parameter([DEVICE_TYPES[CPU], ])
+
+    @run_after('init')
+    def run_after_init(self):
+        """Hooks to run after the init phase"""
+
+        # Filter on which scales are supported by the partitions defined in the ReFrame configuration
+        hooks.filter_supported_scales(self)
+
+        # Make sure that GPU tests run in partitions that support running on a GPU,
+        # and that CPU-only tests run in partitions that support running CPU-only.
+        # Also support setting valid_systems on the cmd line.
+        hooks.filter_valid_systems_by_device_type(self, required_device_type=self.compute_device)
+
+        # Support selecting modules on the cmd line.
+        hooks.set_modules(self)
+
+        # Support selecting scales on the cmd line via tags.
+        hooks.set_tag_scale(self)
+
+    @run_after('init')
+    def set_tag_ci(self):
+        """Set tag CI on smallest benchmark, so it can be selected on the cmd line via --tag CI"""
+        if self.benchmark_info[0] == 'hackathonGPU/benchmark':
+            self.tags.add(TAGS['CI'])
+            log(f'tags set to {self.tags}')
+
+    @run_after('setup')
+    def run_after_setup(self):
+        """Hooks to run after the setup phase"""
+
+        # Calculate default requested resources based on the scale:
+        # 1 task per CPU for CPU-only tests, 1 task per GPU for GPU tests.
+        # Also support setting the resources on the cmd line.
+        if self.compute_device == DEVICE_TYPES[GPU]:
+            hooks.assign_tasks_per_compute_unit(test=self, compute_unit=COMPUTE_UNIT[GPU])
+        else:
+            hooks.assign_tasks_per_compute_unit(test=self, compute_unit=COMPUTE_UNIT[CPU])
+
```

Comment:
You might want to add process binding as well:

```
@run_after('setup')
def set_binding(self):
    hooks.set_compact_process_binding(self)

```

Just to make sure the OS doesn't move the processes across cores/sockets. It potentially makes the performance more reproducible.

Reply:
Solved with https://github.com/EESSI/test-suite/pull/164/commits/06f90c96b9c7e6f01a470a4122cbb4b68117e881